src/Plugin/migrate/process/SkipOnExist.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+namespace Drupal\cardinal_service_profile\Plugin\migrate\process;
+
+use Drupal\Core\Config\ConfigFactoryInterface;
+use Drupal\Core\Database\Connection;
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\migrate\MigrateExecutableInterface;
+use Drupal\migrate\MigrateSkipRowException;
+use Drupal\migrate\ProcessPluginBase;
+use Drupal\migrate\Row;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Migrate process plugin to query the other migrations for identical keys.
+ *
+ * @MigrateProcessPlugin(
+ *   id = "skip_on_exist"
+ * )
+ */
+class SkipOnExist extends ProcessPluginBase implements ContainerFactoryPluginInterface {
+
+  /**
+   * Config Factory Service.
+   *
+   * @var \Drupal\Core\Config\ConfigFactoryInterface
+   */
+  protected $configFactory;
+
+  /**
+   * Database connection service.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $database;
+
+  /**
+   * List of migration ids excluding those defined by the exclude config.
+   *
+   * @var string[]
+   */
+  protected $migrations = [];
+
+  /**
+   * {@inheritDoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('config.factory'),
+      $container->get('database')
+    );
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ConfigFactoryInterface $config_factory, Connection $database) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->configFactory = $config_factory;
+    $this->database = $database;
+  }
+
+  /**
+   * {@inheritDoc}
+   */
+  public function transform($value, MigrateExecutableInterface $migrate_executable, Row $row, $destination_property) {
+    $source_ids = $row->getSourceIdValues();
+    $migration_names = $this->getMigrationConfigs($source_ids);
+
+    foreach ($migration_names as $name) {
+      $query = $this->database->select("migrate_map_$name", 'm')
+        ->fields('m')
+        ->condition('destid1', 0, '>');
+
+      $i = 1;
+      foreach ($source_ids as $key_value) {
+        $query->condition("sourceid$i", $key_value);
+        $i++;
+      }
```

Comment:
I am pretty sure I know what's going on here, but am just a little confused about the $i iteration loop and appending to the sourceid.

Reply:
I'll add some comments to the code. but the table has columns like `sourceid1` and so on depending how many source keys you define in the migration config.

Reply:
Awesome thanks! Looks good to me.